src/main/java/nl/hsac/fitnesse/fixture/slim/web/BrowserTest.java

```diff
@@ -2732,6 +2756,65 @@ public void setNgBrowserTest(NgBrowserTest ngBrowserTest) {
         this.ngBrowserTest = ngBrowserTest;
     }
 
+    public boolean isShadowDomHandlingEnabled(){
+        return shadowDomHandling;
+    }
+
+    public void setShadowDomHandlingTo(boolean shadowDomHandling){
+        this.shadowDomHandling = shadowDomHandling;
+    }
+
+    protected T getElementWithShadow(String cssSelector){
+        try{
+            return (T) shadow.findElement(cssSelector);
+        } catch (Exception e){
+            return null;
+        }
+    }
+
+    private boolean ShadowUsageAllowed(T elementOrContainer) {
```

Comment:
Please make protected and use a method name in line with standard Java conventions (e.g. `isShadowUsageAllowed`). Looking at the body it is not immediately clear to me what the method does/returns.

Reply:
This method checks if all conditions are given before using shadow. e.g. is shadow usage enabled and did fitnesse try to find an element several times the "normal" way before using the shadow plugin? If so and also if it still can't find anything after the 10th try it will return true.

Reply:
Changed it to "isShadowUsageAllowed".